### PR TITLE
Adds warn in email about expiration date

### LINKED ISSUE #2279: Set Email Invitation Expiration Time To A Week for All Casa Admins
**What type(s) of user does this feature affect?**
 - all casa admins

**Description**
A pull request for each of these steps would be great

 - [x] Set the expiration date of the invite token to a week 
 - Write a test checking that the token expires
 - [x] In the account setup email, warn that the link only works for a week
 - Write or modify a test checking that the warning displays the correct date

see https://github.com/scambra/devise_invitable#model-configuration-
> `invite_for`: The period the generated invitation token is valid. After this period, the invited resource won't be able to accept the invitation. When `invite_for` is `0` (the default), the invitation won't expire.

### PR DESCRIPTION
**Pull request by efgalvao**
### What github issue is this PR for, if any?
Resolves #2279

### What changed, and why?
Added warn in e-mail("invitation instructions") about the expiration date of the invitation.

### How will this affect user permissions?
It will not affect.

### How is this tested? (please write tests!) 💖💪
I have created a test to check if the date in the e-mail is the correct date of expiration.

### Screenshots please :)
Screenshot from 2021-07-16 15-44-18

### Feelings gif (optional)
What gif best describes your feeling working on this issue? https://giphy.com/
efgalvao

### CODE REVIEW

**Comment by efgalvao** [comment]
@FireLemons , what you think about creating an exclusive mailer for AllCasaAdmins ?

File: app/views/devise/mailer/invitation_instructions.html.erb [hunk 1]
```diff
@@ -39,6 +39,11 @@
         If you are ready to get started, please set your password. This is the first step to accessing your new account.
       </td>
     </tr>
+    <tr style="font-family: Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; margin: 0;">
+      <td class="content-block" style="font-family: Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; vertical-align: top; margin: 0; padding: 0 0 20px;" valign="top">
```
Comment (brunohkbx) [inline app/views/devise/mailer/invitation_instructions.html.erb:43]: We shouldn't use CSS class in mailers

File: app/views/devise/mailer/invitation_instructions.text.erb [hunk 2]
```diff
@@ -5,5 +5,6 @@
 <% else %>
   A CASA console admin account has been created for you. Your console account is associated with this email.
   If you are ready to get started, please set your password. This is the first step to accessing your new account.
+  <%= t 'devise.mailer.invitation_instructions.accept_until',due_date: l(@resource.invitation_due_at, format: :'devise.mailer.invitation_instructions.accept_until_format') %>
```
Comment (brunohkbx) [inline app/views/devise/mailer/invitation_instructions.text.erb:8]:
```suggestion
  <%= t 'devise.mailer.invitation_instructions.accept_until', due_date: l(@resource.invitation_due_at, format: :'devise.mailer.invitation_instructions.accept_until_format') %>
```

File: lib/mailers/previews/devise_mailer_preview.rb [hunk 3]
```diff
@@ -4,5 +4,9 @@ class DeviseMailerPreview < ActionMailer::Preview
   def reset_password_instructions
     Devise::Mailer.reset_password_instructions(User.first, "faketoken")
   end
+
+  def invitation_instructions
+    Devise::Mailer.invitation_instructions(AllCasaAdmin.first, "faketoken")
+  end
```
Comment (FireLemons) [inline lib/mailers/previews/devise_mailer_preview.rb:10]: thanks, really needed this

File: app/views/devise/mailer/invitation_instructions.html.erb (hunk 1 above)
Comment (FireLemons) [reply app/views/devise/mailer/invitation_instructions.html.erb:43]: it's ok. I don't think the class does anything

File: spec/mailers/casa_admin_mailer_spec.rb [hunk 4]
```diff
@@ -19,4 +19,15 @@
...
   end
+
+  describe ".invitation_instructions for an all casa admin" do
+    let!(:all_casa_admin) { create(:all_casa_admin) }
+    let!(:mail) { all_casa_admin.invite! }
+
+    it "informs the correct expiration date" do
+      expiration_date = all_casa_admin.invitation_due_at.strftime("%B %d, %Y %I:%M %p")
+
+      expect(mail.body.encoded).to match("This invitation will be due in #{expiration_date}")
```
Comment (FireLemons) [inline spec/mailers/casa_admin_mailer_spec.rb:30]: Devise phrased that very strangely. I might change it later.

**Review by FireLemons — APPROVED** [review]

**Pull request merged**